src/Collection.php

```diff
@@ -712,6 +712,11 @@ public function split(int $type = Operation\Splitable::BEFORE, callable ...$call
         return new self(Split::of()($type)(...$callbacks), $this->getIterator());
     }
 
+    public function squash(): CollectionInterface
+    {
+        return self::fromIterable($this->pack()->all())->unpack();
```

Comment:
If you need I can help adding the docs for this method, but would just like to understand a bit better how this works. I'm looking at the docs for `pack` and `unpack` but still don't get why they're needed here 😓.

If you don't mind can you explain what's the difference between this and doing the below?

```PHP
return self::fromIterable($this->all());

Reply:
I suspect it has to do with the collections allowing keys that are not only `int|string` as well as duplicate keys, right?

Reply:
Yes exactly. `iterator_to_array()` will fail if keys are different from `int|string`.

See: https://3v4l.org/FW0Ca